enhancements/alerting/alerting-as-a-feature.md

```diff
@@ -0,0 +1,302 @@
+---
+title: Alerting as a feature
+authors:
+  - "@dofinn"
+reviewers:
+  - "@smarterclayton"
+  - "@jeremyeder"
+  - "@michaelgugino"
+  - "@s-urbaniak"
+  - "@lilic"
+  - "@RiRa12621"
+  - "@wking"
+  - "@cblecker"
+  - "@jharrington22"
+  - "@mwoodson"
+approvers:
+  - TBD
+creation-date: 2021-03-08
+last-updated: 2021-03-08
+status: implementable
+---
+
+# Alerting as a Feature
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [x] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Definitions
+
+Caused Based Alert: Describes the direct cause of an issue. Example: etcdMemoryUtilization @ 100%
+
+Symptom Based Alert = Describes a symptom who's source is a cause based alert. Example: etcd connection latency > 200ms. This may be caused by etcdMemoryUtilization (or not?)
+
+## Summary
+
+Alerting as a feature is a holistic composition of [alerting standards](https://github.com/openshift/enhancements/pull/637) coupled with built-in alerting methodologies (none, symptom based and caused based) with an option to deliver SLOs for a single and/or fleet of clusters. 
+
+## Motivation
+
+OCP as a product does not have built-in [alerting standards](https://github.com/openshift/enhancements/pull/637), nor built-in SLOs driven by symptom or caused based alerting methodologies. This may be acceptable when managing a small number of clusters but at scale (+100) it breaks down. As an SRE of a cluster, you are given alerts and documentation on how to [manage them](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.6/html/monitoring/managing-alerts). This should come "out of the box" to support a natively support fleets.
+
+There are two current efforts in the space at the moment:
+
+* [alerting standards](https://github.com/openshift/enhancements/pull/637) -> this will have component teams own their own components alerts as they are they subject matter experts (caused based alerts). 
```

Comment:
I would be careful here, it sounds like a path to noisy alerts and alert burden. Instead of alerting we might want to invest in tools and practices to drill into such components. As cluster admin, responsible for bigger service (e.g cluster) you should be able to alert on symptoms and one instance of etcd being slow is not a symptom, it might not even impact cluster at all! It is a potentially misleading piece that you should have dashboard for and not screaming alerts 🤔

Reply:
The alerting standards define how component teams will create alerts for their components. They are the experts in their domains. These alerts should aid in troubleshooting a caused based symptom of which an admin/sre can overlay on the cluster. alerting standards being the caused based alerting layer that will live under symptom. IMO this should be life cycled and developed to reduce MTTR.